deploy/7-deploy-pendle-oracle.ts

```diff
@@ -3,50 +3,121 @@ import { DeployFunction } from "hardhat-deploy/dist/types";
   const proxyOwnerAddress = network.live ? ADDRESSES[network.name].timelock : deployer;
-
   const defaultProxyAdmin = await artifacts.readArtifact(
     "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
   );
 
-  const { PTweETH_26DEC2024, PTweETH_26DEC2024_Market, PTOracle, WETH } = ADDRESSES[network.name];
+  const addresses = ADDRESSES[network.name];
+  const fallbackAddress = "0x0000000000000000000000000000000000000001";
 
-  const ptOracleAddress = PTOracle || (await ethers.getContract("MockPendlePtOracle")).address;
+  const deployOracles = async (oracleConfig: OracleConfig) => {
+    for (const { name, market, ptToken, underlyingToken, ptOracle } of oracleConfig) {
+      const ptOracleAddress = ptOracle || (await ethers.getContract("MockPendlePtOracle")).address;
+      await deployPendleOracle(
+        deployments,
+        deployer,
+        name,
+        [market || fallbackAddress, ptOracleAddress, ptToken, underlyingToken, oracle.address, PENDLE_TWAP_DURATION],
+        proxyOwnerAddress,
+        defaultProxyAdmin,
+      );
+    }
+  };
 
-  await deploy("PendleOracle-PT-weETH-26DEC2024", {
-    contract: "PendleOracle",
-    from: deployer,
-    log: true,
-    deterministicDeployment: false,
-    args: [
-      PTweETH_26DEC2024_Market || "0x0000000000000000000000000000000000000001",
-      ptOracleAddress,
-      PTweETH_26DEC2024,
-      WETH,
-      oracle.address,
-      1800,
-    ],
-    proxy: {
-      owner: proxyOwnerAddress,
-      proxyContract: "OptimizedTransparentUpgradeableProxy",
-      viaAdminContract: {
-        name: "DefaultProxyAdmin",
-        artifact: defaultProxyAdmin,
-      },
+  const oracleConfig: OracleConfig = [
+    {
+      name: "PendleOracle-PT-weETH-26DEC2024",
+      market: addresses.PTweETH_26DEC2024_Market,
+      ptToken: addresses.PTweETH_26DEC2024,
+      underlyingToken: addresses.WETH,
+      ptOracle: addresses.newPTOracle || (await ethers.getContract("MockPendleOracle")).address,
     },
-    skipIfAlreadyDeployed: true,
-  });
+    {
+      name: "PendleOracle_PT_USDe_27MAR2025",
+      market: addresses.PTUSDe_27MAR2025_Market,
+      ptToken: addresses.PTUSDe_27MAR2025,
+      underlyingToken: addresses.USDe,
+      ptOracle:
+        network.name === "sepolia"
+          ? (await ethers.getContract("MockPendleOracle_PT_USDe_27MAR2025")).address
+          : addresses.newPTOracle,
+    },
+    {
+      name: "PendleOracle_PT_sUSDe_27MAR2025",
+      market: addresses.PTsUSDe_27MAR2025_Market,
+      ptToken: addresses.PTsUSDe_27MAR2025,
+      underlyingToken: addresses.USDe,
+      ptOracle:
+        network.name === "sepolia"
+          ? (await ethers.getContract("MockPendleOracle_PT_sUSDe_27MAR2025")).address
+          : addresses.newPTOracle,
+    },
+  ];
+
+  await deployOracles(oracleConfig);
+
+  if (network.name === "sepolia") {
+    const NormalTimelock = "0xc332F7D8D5eA72cf760ED0E1c0485c8891C6E0cF"; // SEPOLIA NORMAL TIMELOCK
```

Comment:
In deployment config, in place of timelock we are having guardian address and it will be a complete different task to replace timelock address with the actual ones, IMO.